x/attribute/keeper/keeper.go

```diff
@@ -78,20 +79,24 @@ func (k Keeper) GetAttributes(ctx sdk.Context, acc sdk.AccAddress, name string)
 }
 
 // IterateRecords iterates over all the stored attribute records and passes them to a callback function.
-func (k Keeper) IterateRecords(ctx sdk.Context, handle Handler) error {
+func (k Keeper) IterateRecords(ctx sdk.Context, prefix []byte, handle Handler) error {
 	// Init a attribute record iterator
 	store := ctx.KVStore(k.storeKey)
-	iterator := sdk.KVStorePrefixIterator(store, types.AttributeKeyPrefix)
+	iterator := sdk.KVStorePrefixIterator(store, prefix)
 	defer iterator.Close()
 	// Iterate over records, processing callbacks.
 	for ; iterator.Valid(); iterator.Next() {
 		record := types.Attribute{}
-		if err := types.ModuleCdc.UnmarshalBinaryBare(iterator.Value(), &record); err != nil {
-			return err
+		// get proto objects for legacy prefix with legacy amino codec.
+		if bytes.Equal(types.AttributeKeyPrefix, prefix) {
```

Comment:
Following line is using amino, probably want amino prefix here

Reply:
good catch! :( changed